drivers/sx127x/sx127x_getset.c

```diff
@@ -127,7 +127,7 @@ void sx127x_set_channel(sx127x_t *dev, uint32_t channel)
     /* Save current operating mode */
     dev->settings.channel = channel;
 
-    channel = (uint32_t)((double) channel / (double)LORA_FREQUENCY_RESOLUTION_DEFAULT);
+    channel = (uint32_t)((float) channel / (float)LORA_FREQUENCY_RESOLUTION_DEFAULT);
```

Comment:
I think you can so away with the floating point use entirely.
If you need the precision that is encoded in `LORA_FREQUENCY_RESOLUTION_DEFAULT`, multiply it by a constant factor - but I doubt you need it down to the μHz

Reply:
Yeah I noticed that too, since it gets cast as an int anyway. Would `uint16_t` be sufficient in order to save space or would I need to go up to `uint32_t`?

Reply:
Reading the documentation, [`sx127x_set_channel()`](https://doc.riot-os.org/group__drivers__sx127x.html#ga5df4d694914e0fe575c7a11a25f787be) actually sets the frequency - and then internally calculates the channel from that frequency.

This is the opposite of what all the other radio drivers do.